src/components/organisms/Permission.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import { useWeb3 } from '../../providers/Web3'
+import rbacRequest from '../../utils/rbac'
+import Alert from '../atoms/Alert'
+import Loader from '../atoms/Loader'
+
+export default function Permission({
+  eventType,
+  children
+}: {
+  eventType: string
+  children: ReactElement
+}): ReactElement {
+  const [data, updateData] = useState<boolean>()
+  const { accountId } = useWeb3()
+  useEffect(() => {
+    const getData = async () => {
+      if (accountId !== undefined) {
+        const data = await rbacRequest(eventType, accountId)
+        updateData(data)
+      }
+    }
+    getData()
+  }, [eventType, accountId])
+  if (data === true) {
+    return <>{children}</>
+  } else if (data === false) {
+    const message = `Sorry, you don't have permission to  ${eventType}. Please make sure you have connected your registered address.`
+    return <Alert title="Permission denied" text={message} state="error" />
+  } else {
+    return (
+      <>
+        <Alert
+          text="Please make sure your wallet is connected to proceed."
```

Comment:
based on above logic this message will be also shown when request or response to/from RBAC server fails. Since we should stop just adding more `returns` would suggest rethinking the rendering logic a bit, e.g. always render the `Alert` as fallback but construct and switch its message based on conditions

Reply:
Good idea, I've updated the logic to reduce the return statements and switch the messages like you mention. 

It seems to work alright but the tests are currently failing due to a linting error which I'm not getting locally. Not exactly sure why the linting checks are different as I'm running the same version of ESLint locally: https://github.com/oceanprotocol/market/pull/597/checks?check_run_id=2746525029#step:7:215

Reply:
I solved the linting errors, it seems that ESLint had been upgraded on main since this branch was started

Reply:
@kremalicious let me know if you're happy with the refactoring of the logic to reduce the return statements and switch the alert messages. Also, let me know if you have any other feedback